packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+import getIn from '../../../utils/getIn';
+import type { BaseTextProps } from '../../Typography/BaseText';
+import BaseText from '../../Typography/BaseText';
+import type { Theme } from '../../BladeProvider';
+import { useTheme } from '../../BladeProvider';
+import type { IconComponent, IconProps, IconSize } from '../../Icons';
+import makeSpace from '../../../utils/makeSpace';
+import type { TypographyPlatforms } from '../../../tokens/global/typography';
+import makeBorderSize from '../../../utils/makeBorderSize';
+import StyledBaseButton from './StyledBaseButton';
+import type { ButtonMinHeight, ButtonTypography } from './buttonTokens';
+import { typography } from './buttonTokens';
+
+export type BaseButtonIcon = undefined | null | IconComponent;
+
+type BaseButtonCommonProps = {
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'link';
+  intent?: 'positive' | 'negative' | 'notice' | 'info' | 'neutral';
+  contrast?: 'low' | 'high';
+  size?: 'large' | 'medium' | 'small' | 'xsmall';
+  iconPosition?: 'left' | 'right';
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  onClick?: () => void;
+  type?: 'button' | 'reset' | 'submit';
+};
+
+type BaseButtonWithoutIconProps = BaseButtonCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+type BaseButtonWithIconProps = BaseButtonCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
+
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => ({
+    paddingLeft: hasIcon && iconPosition === 'left' ? iconSpacing : '0px',
+    paddingRight: hasIcon && iconPosition === 'right' ? iconSpacing : '0px',
+  }),
+);
+
+type BaseButtonStyleProps = {
+  iconSize: IconSize;
+  fontSize: keyof Theme['typography']['fonts']['size'];
+  lineHeight: keyof Theme['typography']['lineHeights'];
+  minHeight: ButtonMinHeight;
+  iconSpacing: string;
+  iconColor: IconProps['color'];
+  textColor: BaseTextProps['color'];
+  spacing: string;
+  text?: string;
+  color: string;
+  borderColor: string;
+  hoverColor: string;
+  hoverBorderColor: string;
+  activeColor: string;
+  activeBorderColor: string;
+  focusColor: string;
+  focusBorderColor: string;
+  focusRingColor: string;
+  borderWidth: string;
+  borderRadius: string;
+};
+
+const getProps = ({
+  buttonTypographyTokens,
+  children,
+  isDisabled,
+  size,
+  theme,
+}: {
+  buttonTypographyTokens: ButtonTypography[TypographyPlatforms];
+  children?: string;
+  isDisabled: boolean;
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+}): BaseButtonStyleProps => {
+  const props: BaseButtonStyleProps = {
+    iconSize: 'medium',
+    fontSize: buttonTypographyTokens.fonts.size.medium,
+    lineHeight: buttonTypographyTokens.lineHeights.medium,
+    minHeight: '40px',
```

Comment:
We actually don't. We have `makeSpace`, `makeBorderSize`, `makeTypographySize`, `makeMotionTime`.

Since this was `px` in both platforms, I used it directly. Should we have a `makeDimensionSize`?

Reply:
it can just be `makeSize`